src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,394 @@
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings'])
+    ]).then(answer => {
+      if (answer['confirm'] === 'Pengwings') {
+        // build the configuration
+      }
+    });
+  }
+};
+```
+
+
+## Dev Configs
+
+Congratulations! You've now created the base of an `webpack-scaffold`! Let's now add some more stuff to our future configuration file!
+We are going to follow good convention, and extract our configuration into another file, named `dev-config.js`. As this is just regular JavaScript, we can make the module a function, and supply our `entry` as a parameter for us to build up a configuration file from.
+
+__dev-config.js__
+
+```js
+module.exports = function createDevConfig(answer) {
+  let devConfig = {};
+};
+```
+
+__generator.js__
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+const createDevConfig = require('./dev-config');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings'])
+    ]).then(answer => {
+      if (answer['confirm'] === 'Pengwings') {
+        this.options.env.configuration.dev.webpackOptions = createDevConfig(answer);
+      }
+    });
+  }
+};
+```
+
+We've now abstracted that part of the code that's probably going to be really big. Let's go ahead and add another question, like asking for an entry point.
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+const Input = require('@webpack-cli/webpack-scaffold').Input;
+const createDevConfig = require('./dev-config');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings']),
+      Input('entry', 'What is the entry point in your app?')
+    ]).then(answer => {
+      if (answer['confirm'] === 'Pengwings') {
+        this.options.env.configuration.dev.webpackOptions = createDevConfig(answer);
+      }
+    });
+  }
+};
+```
+
+Let's go ahead and create our config.
+
+
+## Some more configs
+
+Let's start by looking at `dev-config.js`. We have some answers, now we want to use them to build up a config. We will start with mounting our values on the variables we've declared, with some properties we want to build up. We also want to use the answers for the `entry` property. We've also added an output property that has a `filename`.
```

Comment:
I've been reading this a few times so far, it still feels wrong. Maybe lets go with:
```suggestion
Let's look at `dev-config.js`. We have access to user's answers, use them to assign values to desired config properties, in this case - `entry`. We've also added an output property that has a `filename`.
```